src/Workspaces/Remote/ServiceHub/Shared/Extensions.cs

```diff
@@ -73,7 +74,7 @@ internal static partial class Extensions
 
                 // record reason why task got aborted. use NFW here since we don't want to
                 // crash VS on explicitly killing OOP.
-                task.Exception.ReportServiceHubNFW("JsonRpc Invoke Failed");
+                (task?.Exception ?? ex).ReportServiceHubNFW("JsonRpc Invoke Failed");
```

Comment:
why not always just report whatever 'ex' we get back.  seems odd to ever use `task.Exception` tbh.

Reply:
what's teh difference ebtween reporting  a service hub NFW, and throwing a SoftCrash exception?  do we need both systems?

Reply:
comments apply to the duplicated cases below.